pkg/alertmanager/api_grafana.go

```diff
@@ -309,8 +309,25 @@ func (am *MultitenantAlertmanager) SetUserGrafanaConfig(w http.ResponseWriter, r
 		return
 	}
 
-	payload, err := io.ReadAll(r.Body)
+	var input io.Reader
+	maxConfigSize := am.limits.AlertmanagerMaxGrafanaConfigSize(userID)
+	if maxConfigSize > 0 {
+		input = http.MaxBytesReader(w, r.Body, int64(maxConfigSize))
+	} else {
+		input = r.Body
+	}
+
+	payload, err := io.ReadAll(input)
 	if err != nil {
+		maxBytesErr := &http.MaxBytesError{}
+		if errors.As(err, &maxBytesErr) {
```

Comment:
I think actually because you don't use `maxBytesErr`, we can just do `Is()`

```suggestion
		if errors.Is(err, http.MaxBytesError) {
```
```

Reply:
I'd rather use `errors.Is` also, but in this case it doesn't work since `http.MaxBytesError` is a struct 😢 
![Screenshot 2024-09-25 at 13 49 06](https://github.com/user-attachments/assets/b6165c3a-8c2b-4e78-9265-65ebefb3561a)

Reply:
You can make it a one-liner though:
```suggestion
		if maxBytesErr := (&http.MaxBytesError{}); errors.As(err, &maxBytesErr) {
```

Reply:
Ahh nice, I was tried something like `errors.As(err, &(&http.MaxBytesError{}))` before - forgot I could have a short statement there. Tks for the tip 🙏